Pieces/Piece.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Chess.Board;
+namespace Chess.Pieces
+{
+    abstract class Piece
+    {   
+        protected bool first_move = true;
```

Comment:
G24. Sugeruje używanie konwencji nazewnictwa jak w linku https://docs.microsoft.com/pl-pl/dotnet/standard/design-guidelines/general-naming-conventions - czyli brak `underscora`, wielkie litery (reszte tak samo, juz nie bede komentować), 

dodatkowo N4. nazwa jest myląca, `first_move = true` można interpretować ze wykonał już pierwszy ruch jak i ze dopiero wykona, proponuję jak poniżej lub podobnie: 
```suggestion
        protected bool FirstMoveExecuted = false;
```

Reply:
Poprawiono